docs/DeveloperGuide.md

```diff
@@ -1,25 +1,277 @@
+
+## Implementation - Features
+
+### Print to PowerPoint SlideShow
+
+An additional feature targeted at students who wish to use add more style to their flash cards outside of the command 
+line option to allow keep things interesting when they are revising.
+
+The `PowerPointCommand` is parsed by `NormalParser` but the "Print to PowerPoint" command can be called from both Normal 
+and Deck Mode. 
+
+The following diagram shows how the `PowerPointCommand`'s `execute()` calls the `createNewPowerPoint()` method of `PowerPoint`.
+![PPTX Sequence Diagram](images-dg/DG-PPTX-Sequence-Diagram.png)
+
+The `newIntroSlide()`, `newSlide()` and `exportSlide()` method of `PowerPoint` uses a third party library - Apache POI 
+to create new slides, populate them with the questions and answers from the deck and finally print them out to a new PowerPoint
+ file in the `pptx` folder under the name `<deck name>.pptx`.
+ 
+The following are the Classes/ Enum of the third part package `org.apache.poi.xslf.usermodel` which are used:
+* `SlideLayout` - Enum representing the Slide Layouts available
+* `XMLSlideShow` - Class representing an entire Slide Show
+* `XSLFSlide` - Class representing a single Slide
+* `XSLFSlideLayout` - Class representing the layout of a slide
+* `XSLFSlideMaster` - Class representing the default slides layouts
+* `XSLFTextShape` - Class representing a shape within a slide
+* `XSLFTextParagraph` - Class representing a paragraph of text within a shape
+* `XSLFTextRun` - Class representing the properties of the text within a paragraph
+
+### Pretty Printing (Wei Siew)
+
+The purpose of this feature is to improve the readability of the
+question and answer fields of a flashcard for the user. Without this
+feature, long question and answer fields will follow the default
+wrapping style of the console. When words are truncated unnecessarily,
+it is going to be distracting and annoying for students trying to
+study. We illustrate the problem in the following example:
+```
+This is a long question (or maybe answer) field. Suppose tha
+t our console is 60 characters wide, we see that the word "t
+hat" was truncated in the first line and again in the second
+line.
+```
+In this section, we define the following terms:
+* `lineLength` is the maximum number of characters on a line,
+set to be equal to Ui.DASH_LINES.length(). This is also the number of
+characters between the start of line and end of line.
+* `label` can be "Question: " or "Answer:   " and is used to indicate
+whether a field is the question or answer of the flashcard.
+* `usableLength` is the number of characters that can be used for
+printing a field. This is also the number of characters between the end
+of `label` and end of line.
+
+The following sequence diagram illustrates the call to the
+`toString(boolean isQuestion, int offset)` method of a `FlashCard`
+object.
+
+![DG-Implementation-Features-PP-Sequence](./images-dg/PP-Sequence.png?raw=true)
```

Comment:
Are you labelling correct? Perhaps it should be toString(isQuestion, offset) and formatResponse(isQuestion, offset)

![image](https://user-images.githubusercontent.com/60382274/97516241-aa3c4a00-19cd-11eb-8e3e-5718f4eec84a.png)